zebrad/src/components/sync.rs

```diff
@@ -149,18 +129,31 @@ where
     ///  - verifier: the zebra-consensus verifier that checks the chain
     pub fn new(chain: Network, peers: ZN, state: ZS, verifier: ZV) -> Self {
         let tip_network = Timeout::new(peers.clone(), TIPS_RESPONSE_TIMEOUT);
-        let block_network = ServiceBuilder::new()
-            .retry(zn::RetryLimit::new(BLOCK_DOWNLOAD_RETRY_LIMIT))
-            .timeout(BLOCK_DOWNLOAD_TIMEOUT)
-            .service(peers);
-        let verifier = Timeout::new(verifier, BLOCK_VERIFY_TIMEOUT);
+        // The Hedge middleware is the outermost layer, hedging requests
+        // between two retry-wrapped networks.  The innermost timeout
+        // layer is relatively unimportant, because slow requests will
+        // probably be pre-emptively hedged.
+        //
```

Comment:
```suggestion
        // layer is relatively unimportant, because slow requests will
        // probably be pre-emptively hedged.
        //
        // The Hedge goes outside the Retry, because the Retry layer
        // abstracts away spurious failures from individual peers
        // making a less-fallible network service, and the Hedge layer
        // tries to reduce latency of that less-fallible service.
        //
```